.github/workflows/main.yml

```diff
@@ -0,0 +1,171 @@
+      with:
+        versionSpec: '5.3.6'
+      env: 
+        ACTIONS_ALLOW_UNSECURE_COMMANDS: TRUE
+    
+    - name: Determine Version      
+      uses: gittools/actions/gitversion/execute@v0.10.2
+      with:
+        useConfigFile: true        
+      env: 
+        ACTIONS_ALLOW_UNSECURE_COMMANDS: TRUE
+
+    - name: get SemVer       
+      id: gitversion
+      run: |
+        echo "SemVer: v$($env:GitVersion_SemVer)" 
+        Write-Host "::set-output name=semver::v$($env:GitVersion_SemVer)"                
+        $fileName = "dummy-v$($env:GitVersion_SemVer)"
+        echo "SemVerMMP: v$($env:GitVersion_MajorMinorPatch)"
+        Write-Host "Filename: '$fileName'" 
+
+
+    - name: Setup msbuild
+      uses: microsoft/setup-msbuild@v1.1
+      with:
+          vs-version: '12.0'
+     
+      
+    - name: Restore Packages
+      run: Nuget restore SourceGrid.sln
+
+    - name: Build
+      run: msbuild -m -t:Rebuild /p:Configuration=Release /p:Platform="Any CPU" /fl SourceGrid.sln
+        
+    
+    - name: Archive Build Log
+      uses: actions/upload-artifact@v2
+      if: ${{ success() || failure() }}
+      with:
+        name: Compile_Solution_log
+        path: |
+          *.log    
+    
+      
+      
+    - name: Create zip
+      id: packageBuildResults
+      run: |
+        $sourceFolder = Join-Path $env:GITHUB_WORKSPACE "out" | Join-Path -ChildPath "net*" 
+        $outFolder = Join-Path $env:GITHUB_WORKSPACE "out" | Join-Path -ChildPath "sourcegrid"
+        New-Item -ItemType Directory -Force -Path $outFolder
+        $fileName = "sourcegrid-2.0.0.zip"
+        Write-Host "Filename: '$fileName'"
+        Write-Host "sourceFolder: '$sourceFolder'"
+        Write-Host "Outfolder: '$outFolder'"
+        Write-Host "::set-output name=sourcegrid::$($fileName)"
+        Remove-Item $sourceFolder\**\*.pdb -Force -Recurse
+        $outPath = Join-Path $outFolder $fileName
+        Compress-Archive -DestinationPath $outPath -Path $sourceFolder -CompressionLevel Optimal
+      
+        
+    - name: Archive Build Output
+      uses: actions/upload-artifact@v2
+      with:
+         name: sourcegrid
+         path: |
+            out/sourcegrid
+
+
+    - name: Create Nuget Packages 
+      id: createNupkg
+      run: |
+        nuget pack .nuget\SourceGrid.nuspec -Version 2.0.0
+        Write-Host "::set-output name=nupkg-sourcegrid::sourcegrid.2.0.0.nupkg"
+        
+    - name: Archive NuGet Packages
+      uses: actions/upload-artifact@v2
+      with:
+        name: nuget-sourceGrid
+        path: |
+          *.nupkg
+
+  release:
+    if: github.ref == 'refs/heads/Github_runner'    
+    runs-on: windows-latest
+    needs: build
+  
+    steps:
+      - name: Download Build Output
+        uses: actions/download-artifact@v2
+        with:
+          name: sourcegrid
+
+      - name: Download NuGet Package
+        uses: actions/download-artifact@v2
+        with:
+          name: nuget-sourceGrid
+
```

Comment:
what it does ?

Reply:
It will download the build artifacts from previous stage to this stage